src/Rectangle.js

```diff
@@ -54,6 +54,105 @@ class Rectangle {
         }
     }
 
+    /**
+     * Initialize a Rectangle to a 1 unit square transformed by a model matrix.
+     * @param {Array.<number>} m A 4x4 matrix to transform the rectangle by.
+     */
+    initFromModelMatrix (m) {
+        // Treat this function like we are transforming a vector with each
+        // component set to 0.5 by a matrix m.
+        // const v0 = 0.5;
+        // const v1 = 0.5;
+        // const v2 = 0.5;
+
+        // Of the matrix to do this in 2D space, instead of the 3D provided by
+        // the matrix, we need the 2x2 "top left" that represents the scale and
+        // rotation ...
+        const m00 = m[(0 * 4) + 0];
+        const m01 = m[(0 * 4) + 1];
+        const m10 = m[(1 * 4) + 0];
+        const m11 = m[(1 * 4) + 1];
+        // ... and the 1x2 "top right" that represents position.
+        const m30 = m[(3 * 4) + 0];
+        const m31 = m[(3 * 4) + 1];
+
+        // This is how we would normally transform the vector by the matrix.
```

Comment:
This whole description, because it's in a giant comment block is actually pretty hard to read.

I wonder if we shouldn't maintain this in like `docs/Rectangle-AABB-Math.md` or some other format that could give us a better formatted view here...

I first suggested to Z to just post this whole explaination into a gist and link it here for reference later, as this wouldn't be so hard to update as you can fork/edit gists.